tests/chart_tests/test_byo_sa.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+
+from tests import supported_k8s_versions
+from tests.chart_tests.helm_template_generator import render_chart
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+class TestServiceAccounts:
+    def test_SAs_created(self, kube_version):
+        # Render the chart with rbacEnabled set to False
+        docs = render_chart(kube_version=kube_version, values={"global": {"rbacEnabled": False}})
+
+        # Check that no ServiceAccount resources are created
+        """
+        service_accounts = [
+            doc["metadata"]["name"] for doc in docs if isinstance(doc, dict) and doc.get("kind") == "ServiceAccount"
+        ]
+        assert len(service_accounts) == 0, "No ServiceAccounts should be created when rbacEnabled is False"
+        """
+        # Check that the Deployment or StatefulSet is using the default ServiceAccount
+        for doc in docs:
+            if isinstance(doc, dict) and doc.get("kind") in ["Deployment", "StatefulSet"]:
+                spec = doc.get("spec", {}).get("template", {}).get("spec", {})
+                sa_name = spec.get("serviceAccountName", "default")
+                assert sa_name == "default", f"Expected default ServiceAccount, but got {sa_name}"
```

Comment:
These `.get()` are not going to prevent the assertion from failing, so we don't really need them. We can just assume that the data structure is what we hope to find, because we'll encounter some kind of failure if it's not.

Also we don't want to insert a default value into the tests. We want to see whatever comes out of the helm chart. If we need a default, that belongs in the helm template, not the test case.

```suggestion
                assert (
                    sa_name == doc["spec"]["template]"]["spec"]["serviceAccountName"]
                ), f"Expected default ServiceAccount, but got {sa_name}"
```

Reply:
done